stablehlo/reference/Ops.cpp

```diff
@@ -648,6 +660,45 @@ Tensor evalRealOp(const Tensor &operand, ShapedType resultType) {
   return result;
 }
 
+SmallVector<Tensor> evalReduceOp(ArrayRef<Tensor> inputs,
+                                 ArrayRef<Tensor> initValues,
+                                 const Axes &dimensions, Region &body,
+                                 Scope &scope,
+                                 ArrayRef<ShapedType> resultTypes) {
+  SmallVector<Tensor> results;
+  for (auto [resultType, initValue] : llvm::zip(resultTypes, initValues)) {
+    Tensor result(resultType);
+    for (auto resultIt = result.index_begin(); resultIt != result.index_end();
+         ++resultIt)
+      result.set(*resultIt, initValue.get({}));
+    results.push_back(result);
+  }
+
+  for (auto inputIt = inputs[0].index_begin(); inputIt != inputs[0].index_end();
+       ++inputIt) {
+    Index resultIndex = *inputIt;
+    for (auto dim : llvm::reverse(dimensions))
+      resultIndex.erase(resultIndex.begin() + dim);
```

Comment:
I don't think this is going to work as is because `dimensions` are not guaranteed to be sorted.

Reply:
How about rewriting in a less fancy style? This doesn't increase the size of the code, but is easier to understand and maps directly on what the spec says: "shape(results[k]) = shape(inputs[k]) except that the dimension sizes of inputs[k] corresponding to dimensions are not included".

```C++
    Index resultIndex;
    for (auto [inputAxis, inputIndexElement] : llvm::enumerate(*inputIt)) {
      if (llvm::is_contained(dimensions, inputAxis)) continue;
      resultIndex.push_back(inputIndexElement);
    }
```

Reply:
Thanks, this looks more intuitive.